packages/api/src/external/carequality/document/process-document-query-results.ts

```diff
@@ -82,86 +122,60 @@ export async function processDocumentQueryResults({
   }
 }
 
-function combineDocRefs(documentQueryResults: DocumentQueryResult[]): DocumentReference[] {
-  return documentQueryResults.reduce((acc: DocumentReference[], curr) => {
-    const documentReferences = curr.data.documentReference ?? [];
-
-    return [...acc, ...documentReferences];
-  }, []);
-}
-
-const getNonExistentDocRefs = async (
-  documents: DocumentReference[],
-  patientId: string,
-  cxId: string
-): Promise<DocumentReference[]> => {
-  const [{ existingDocRefs, nonExistingDocRefs }, FHIRDocRefs] = await Promise.all([
-    filterOutExistingDocRefsS3(documents, patientId, cxId),
-    getDocRefsFromFHIR(cxId, patientId),
-  ]);
-
-  const fhirDocRefIds = new Set(FHIRDocRefs.map(d => d.id));
-
-  const foundOnStorageButNotOnFHIR = existingDocRefs.filter(f => !fhirDocRefIds.has(f.docUniqueId));
-
-  const docsToDownload = nonExistingDocRefs.concat(foundOnStorageButNotOnFHIR);
-
-  return docsToDownload;
-};
-
-type ExistentialDocRefs = {
-  existingDocRefs: DocumentReference[];
-  nonExistingDocRefs: DocumentReference[];
-};
-
-const filterOutExistingDocRefsS3 = async (
-  documents: DocumentReference[],
-  patientId: string,
-  cxId: string
-): Promise<ExistentialDocRefs> => {
-  const docIdWithExist = await Promise.allSettled(
-    documents.map(async (doc): Promise<{ docId: string; exists: boolean }> => {
-      const fileName = createS3FileName(cxId, patientId, doc.docUniqueId);
-
-      const { exists } = await s3Utils.getFileInfoFromS3(fileName, s3BucketName);
-
-      return {
-        docId: doc.docUniqueId,
-        exists,
-      };
-    })
-  );
-
-  const successfulDocs = docIdWithExist.flatMap(ref =>
-    ref.status === "fulfilled" && ref.value ? ref.value : []
-  );
-
-  const existentialDocRefs = documents.reduce(
-    (acc: ExistentialDocRefs, curr) => {
-      for (const succDoc of successfulDocs) {
-        if (succDoc.docId === curr.docUniqueId) {
-          if (succDoc.exists) {
-            acc.existingDocRefs = [...acc.existingDocRefs, curr];
-          } else {
-            acc.nonExistingDocRefs = [...acc.nonExistingDocRefs, curr];
-          }
-        }
+// Since we have most of the document contents when doing the document query,
+// we will store this in s3 and then upsert the reference to the s3 object in FHIR
+// when doing the doc retrieval
+async function storeInitDocRefInFHIR(
+  docRefs: DocumentWithMetriportId[],
+  cxId: string,
+  patientId: string
+) {
+  await Promise.allSettled(
+    docRefs.map(async docRef => {
```

Comment:
Wondering if we should use `executeAsynchronously(...)` and limit the amount of parallel requests to S3. It has a fairly large limit ([5,500 req/s prefix - clientId](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html)), but it might answer with 503 (slow down) while it scales to it. To avoid having to code around the 503, we could just limit to 500 async requests?

Reply:
I can add this because i think it makes sense either way but to clarify this is upserting to the FHIR server

Reply:
I set 500 but let me know if you think there is a more apporpriate number considering the above

Reply:
Is [this](https://github.com/metriport/metriport/pull/1437/files/7ffc6ec89c86d810dfbe66a71f70c28abb519abb..2b99e07393ee12336aec40ffb5acbf4aaed7e6fd#diff-344dac11dfe753defa52228f635bcf845e37eeca961219c324c49db18fea809dR121-R137) the code we were referring to? If so, we're not using `executeAsynchronously` and just parallelizing all queries to S3 w/o any constraints.

Maybe you mixed this with the [parallel upserts to FHIR](https://github.com/metriport/metriport/pull/1437/files/7ffc6ec89c86d810dfbe66a71f70c28abb519abb..2b99e07393ee12336aec40ffb5acbf4aaed7e6fd#diff-410886a78a3c700427fa6fed200872ed2130e65c445bb421d11775f07a18084bR26)?

Reply:
It may have been ill update the query to s3 as well to use `executeAsynchronously`